lib/DataFileUtil/DataFileUtilImpl.py

```diff
@@ -122,10 +122,25 @@ def _pack(self, file_path, pack):
         # TODO is there a designated temp files dir in the scratch space? Nope.
         (fd, tf) = tempfile.mkstemp(dir=self.tmp)
         os.close(fd)
-        ctf = shutil.make_archive(tf, arch, d)
+        if arch == 'gztar':
+          ctf = shutil.make_archive(tf, arch, d)
+          suffix = ctf.replace(tf, '', 1)
+          shutil.move(ctf, file_path + suffix)
+        elif arch == 'zip':
+          suffix  = '.zip'
+          with zipfile.ZipFile(tf + suffix, 'w',
+                        zipfile.ZIP_DEFLATED,
+                        allowZip64=True) as zip_file:
+              for root, dirs, files in os.walk(d):
+                for file in files:
+                  zip_file.write(os.path.join(root, file), file)
+
+          shutil.move(tf + suffix, file_path + suffix)
+        else:
+          raise ValueError('Unexpected archive type: {}'.format(arch))
```

Comment:
This can't happen - the pack variable is checked at the beginning of the method and arch is 1:1 with pack.
In fact, with the new logic you could just get rid of the arch variable altogether.